### PR TITLE
Update docs and requirements for dask-expr and remove warning

### PR DESCRIPTION
**Pull request by phofl**

### CODE REVIEW

File: docs/source/api.rst [hunk 1]
```diff
@@ -4,7 +4,7 @@ API Reference
 Dask APIs generally follow from upstream APIs:
 
 -  :doc:`Arrays<array-api>` follows NumPy
--  :doc:`DataFrames <dataframe-api>` follows Pandas
+-  :doc:`DataFrames <dask-expr-api>` follows Pandas
```
Comment (mrocklin) [inline docs/source/api.rst:7]: Maybe we keep the same URL, but change the content?

Comment (phofl) [reply docs/source/api.rst:7]: Hm good point, that makes sense

**Comment by hendrikmakait** [comment]
`test_groupby_dataframe_cum_caching` isn't happy, is that related?

**Review by hendrikmakait — APPROVED** [review]
Thanks, @phofl! `test_groupby_dataframe_cum_caching` failed once, I'm not sure if that's related. Anyhow, I'm inclined to merge this to unblock `distributed` CI regardless.

**Pull request merged**